### PR TITLE
add coverage

### PR DESCRIPTION
**Pull request by khaihkd**

### CODE REVIEW

**Comment by thanhson1085** [comment]
cool, should ignore coverage.json add add travis and coverage badge to readme file

**Comment by thanhson1085** [comment]
We should not push 'coverage ui' to repo.

**Comment by etienne-napoleone** [comment]
Can we have the coverage run without UI?
solidity-coverage is the package used?

File: coverage.json [hunk 1]
```diff
@@ -0,0 +1 @@
...e":45,"column":8}}]},"4":{"line":50,"type":"if","locations":[{"start":{"line":50,"column":8},"end":{"line":50,"column":8}},{"start":{"line":50,"column":8},"end":{"line":50,"column":8}}]},"5":{"line":51,"type":"if","locations":[{"start":{"line":51,"column":8},"end":{"line":51,"column":8}},{"start":{"line":51,"column":8},"end":{"line":51,"column":8}}]},"6":{"line":56,"type":"if","locations":[{"start":{"line":56,"column":8},"end":{"line":56,"column":8}},{"start":{"line":56,"column":8},"end":{"line":56,"column":8}}]},"7":{"line":61,"type":"if","locations":[{"start":{"line":61,"column":8},"end":{"line":61,"column":8}},{"start":{"line":61,"column":8},"end":{"line":61,"column":8}}]},"8":{"line":66,"type":"if","locations":[{"start":{"line":66,"column":8},"end":{"line":66,"column":8}},{"start":{"line":66,"column":8},"end":{"line":66,"column":8}}]},"9":{"line":107,"type":"if","locations":[{"start":{"line":107,"column":8},"end":{"line":107,"column":8}},{"start":{"line":107,"column":8},"end":{"line":107,"column":8}}]},"10":{"line":163,"type":"if","locations":[{"start":{"line":163,"column":12},"end":{"line":163,"column":12}},{"start":{"line":163,"column":12},"end":{"line":163,"column":12}}]}}}}
```
Comment (etienne-napoleone) [inline coverage.json]: This file contains absolute path like `/Users/khaihkd/projects/tomo/tomo-master/contracts/BlockSigner.sol`
Either this is a generated file that should not be versioned, either it should not contain this path

File: package.json [hunk 2]
```diff
@@ -14,25 +14,27 @@
...
-    "homepage": "https://github.com/tomochain/tomo-governance#readme",
+    "homepage": "https://github.com/tomochain/tomomaster#readme",
     "dependencies": {
         "bignumber.js": "^7.2.1",
         "body-parser": "^1.18.2",
         "config": "^1.30.0",
         "express": "^4.16.3",
         "express-validator": "^5.1.2",
         "mongoose": "^5.0.17",
+        "solidity-coverage": "^0.5.4",
```
Comment (etienne-napoleone) [inline package.json]: It should be a dev dependencie

**Comment by etienne-napoleone** [comment]
In my opinion we should just add `solidity-coverage` as a dev package and the `.solcover.js` file to contain it's configuration.
But the files generated by the command should not be versioned! the .gitignore can be edited accordingly

This would still five us information on coverage in console when running in travis. if it fails or someone need more information about it, just run it locally.
The HTML generated must be seen as a prettier output than the console one and therefore must not be versioned.

**Comment by khaihkd** [comment]
@etienne-napoleone I move new package to devDependency + remove coverage result (html & json file). Please review

File: .coveralls.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+repo_token:
```
Comment (etienne-napoleone) [inline .coveralls.yml]: I think this file is not necessary. From the doc:
> This is only needed for private repos and should be kept secret – anyone could use it to submit coverage data on your repo’s behalf.

File: .travis.yml [hunk 4]
```diff
@@ -7,6 +7,8 @@ before_install:
 - if [[ `npm -v` != 6* ]]; then npm i -g npm@6; fi
 install:
 - npm install
+- echo $COVERALLS_TOKEN >> .coveralls.yml
```
Comment (etienne-napoleone) [inline .travis.yml]: Same comment here

File: .coveralls.yml (hunk 3 above)
Comment (khaihkd) [reply .coveralls.yml]: No. this file is required to generate coverage badge number. We use `coveralls` package, this file is storage a token and we use this token to upload coverage to their system. You can see our coverage badge in here https://coveralls.io/github/tomochain/tomomaster?branch=master

Comment (etienne-napoleone) [reply .coveralls.yml]: I feel like it's needed when run locally but not on a CI like Travis:
https://github.com/nickmerwin/node-coveralls#running-locally
I'm not sure that we need, can we just try to change it see if it still work without it?

Comment (etienne-napoleone) [reply .coveralls.yml]: Ah nice, so we can remove it as it's not needed in the final workflow right? :+1:

Comment (khaihkd) [reply .coveralls.yml]: You can see it in my repo https://github.com/khaihkd/tomomaster on the top README file.

Comment (etienne-napoleone) [reply .coveralls.yml]: Ok, anyway, we can change it later if it's not needed

**Pull request merged**